profiling/benches/interning_strings.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use criterion::*;
+use datadog_profiling::collections::string_table::{
+    wordpress_test_data::WORDPRESS_STRINGS, StringTable,
+};
+
+pub fn small_wordpress_profile(c: &mut Criterion) {
+    c.bench_function("benching string interning on wordpress profile", |b| {
+        b.iter(|| {
+            let mut table = StringTable::new();
+            let n_strings = WORDPRESS_STRINGS.len();
+            for string in WORDPRESS_STRINGS {
+                black_box(table.intern(string));
+            }
+            assert_eq!(n_strings, table.len());
+
+            // re-insert, should nothing should be inserted.
+            for string in WORDPRESS_STRINGS {
+                black_box(table.intern(string));
+            }
+            assert_eq!(n_strings, table.len())
+        })
+    });
+}
```

Comment:
I'm curious: did you get a chance to compare this with the previous implementation?

Reply:
I haven't yet, but intend to!

Reply:
Alright, some numbers! Changing from the old implementation to the new one resulted in:

```
benching string interning on wordpress profile
                        time:   [53.695 µs 53.977 µs 54.279 µs]
                        change: [-28.080% -27.464% -26.850%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 3 outliers among 100 measurements (3.00%)
  1 (1.00%) low mild
  2 (2.00%) high severe
```

And then changing it back resulted in:

```
benching string interning on wordpress profile
                        time:   [75.224 µs 75.501 µs 75.820 µs]
                        change: [+40.269% +41.386% +42.503%] (p = 0.00 < 0.05)
                        Performance has regressed.
```

This is unsurprising, because we spend less time in allocators, but I'm glad to have some data!